src/dtsh/cli.py

```diff
@@ -98,6 +119,24 @@ def theme(self) -> Optional[str]:
         """Additional styles file."""
         return str(self._argv.theme[0]) if self._argv.theme else None
 
+    @property
+    def batch_source(self) -> Optional[Union[str, List[str]]]:
+        """Batch command source, if defined."""
+        if self._argv.c:
+            return cast(List[str], self._argv.c)
+        elif self._argv.f:
+            return self._argv.f
+        else:
+            return None
```

Comment:
Could we:
- fix the *unnecessary elif after return*
- type cast the string too 

```suggestion
        if self._argv.c:
            return cast(List[str], self._argv.c)
        if self._argv.f:
            return cast(str, self._argv.f)
        return None
```

All these `cast()` are a bit ugly, but IIUC it's the recommended approach for dealing with non typed modules within strictly typed source code:
- like all Python type hinting, `cast()` is for linters only, and should eventually vanish
- *implicit type casting from Any*, e.g. `cmds: List[str] = self._argv.c`, is (subjectively) less ugly, but creates a new variable (optmized out ?)
- and these `str()` and `bool()` I used in other properties, e.g. `return bool(self._argv.user_files)`, are even more ugly and questionable (I'll fix them)